src/main/java/org/codeseoul/event_member_management/member/MemberController.java

```diff
@@ -33,49 +36,84 @@ public class MemberController {
-                .orElseGet(() -> {
-                    member.setId(id);
-                    return assembler.toModel(repository.save(member));
-                });
+    EntityModel<Member> replaceMember(@Valid @RequestBody Member newMember, @PathVariable Long id, BindingResult errors) throws MethodArgumentNotValidException {
+        Optional<Member> dbMember = repository.findById(id);
+        if (dbMember.isPresent()) {
+            Member oldMember = dbMember.get();
+            validateExistingMember(oldMember, newMember, errors);               
+            oldMember.setUsername(newMember.getUsername());
+            oldMember.setDisplayName(newMember.getDisplayName());
+            oldMember.setEmail(newMember.getEmail());
+            oldMember.setPhoneNumber(newMember.getPhoneNumber());
+            oldMember.setImageUrl(newMember.getImageUrl());
+            oldMember.setShortBio(newMember.getShortBio());
+            return assembler.toModel(repository.save(oldMember));
+        } else {
+            throw new MemberNotFoundException(id);
+        }
     }
 
     @DeleteMapping("/members/{id}")
     void deleteMember(@PathVariable Long id) {
+        repository.findById(id)
+            .orElseThrow(() -> new MemberNotFoundException(id));
         repository.deleteById(id);
     }
+
+    private void validateExistingMember(Member oldMember, Member newMember, BindingResult errors) throws MethodArgumentNotValidException {
+        if (!oldMember.getUsername().equals(newMember.getUsername()) && repository.findByUsername(newMember.getUsername()).isPresent())
+            errors.addError(new FieldError("Member", "username", "username already in use"));
+        if (!oldMember.getEmail().equals(newMember.getEmail()) && repository.findByEmail(newMember.getEmail()).isPresent())
+            errors.addError(new FieldError("Member", "email", "email already in use"));
+        if (!oldMember.getPhoneNumber().equals(newMember.getPhoneNumber()) && repository.findByPhoneNumber(newMember.getPhoneNumber()).isPresent())
+            errors.addError(new FieldError("Member", "phoneNumber", "phoneNumber already in use"));
+        if (errors.hasErrors())
+            throw new MethodArgumentNotValidException(null, errors);
+    }
+
+    private void validateNewMember(Member member, BindingResult errors) throws MethodArgumentNotValidException {
+        if (repository.findByUsername(member.getUsername()).isPresent())
+            errors.addError(new FieldError("Member", "username", "username already in use"));
+        if (repository.findByEmail(member.getEmail()).isPresent())
+            errors.addError(new FieldError("Member", "email", "email already in use"));
+        if (repository.findByPhoneNumber(member.getPhoneNumber()).isPresent())
+            errors.addError(new FieldError("Member", "phoneNumber", "phoneNumber already in use"));
+        if (errors.hasErrors())
+            throw new MethodArgumentNotValidException(null, errors);
```

Comment:
I think this could be used by `validateExistingMember`. I'd say use the new repository method I described above, and have the existing member method just call this one in addition to comparing to `oldMember`.

Reply:
That's what I did originally, but because I wanted to limit the number of queries I deferred the check. If we can figure out how to make a cleaner lookup it will be easier to reuse.